### DISCUSSION TITLE
Ruby 3.1 WIP

### DISCUSSION THREAD

**Discussion by pderouen** [Origen-SDK/origen#437] [Show and tell]
@priyavadan I have started a [branch](https://github.com/Origen-SDK/origen/tree/ruby31_wip) to get Origen running with Ruby 3.1. This was done from Windows not Linux. Getting Origen to install was a little weird. I couldn't get gem install to install anything newer than 0.34.0 of origen.

- Lint is still broken (so sorry about the poor formatting)
- 1 spec test is still reporting a failure (ERB crashes on compile)

**Comment by priyavadan** [comment]
Thanks, I tried your origen branch with ruby 3.0.2  on my linux environment

- origen specs is passing
- origen test is passing

Concern about updating origen_testers to use ruby 3.* as pin  API was broken when I last tried it a few months back.

![image](https://user-images.githubusercontent.com/3895377/164545935-d69040c3-9b58-4472-a24e-3ffcd147749c.png)

![image](https://user-images.githubusercontent.com/3895377/164546016-139ba0d4-462c-49f8-9cee-6be2982fa6b5.png)


![image](https://user-images.githubusercontent.com/3895377/164546032-a93286c0-d1cd-4f8e-99cf-473b7864001d.png)

**Reply by pderouen** [reply]
Nice! You should give origen testers a try with this ruby 3 branch. I had to make several changes to get the pins specs passing. Worth a check to see if those updates resolve the issue with testers.

**Reply by priyavadan** [reply]
I just tried, still seeing failures --> haven't diven deep into it yet but snot sure how early I can get on to it, yet.

![image](https://user-images.githubusercontent.com/3895377/164548503-2e8cc0b8-cc6f-4cff-9b58-51e8a2ec73e9.png)

**Reply by pderouen** [reply]
That looks very similar to errors I saw. I had to change the way a few methods defined arguments. It doesn't like the "**whatever" anymore. It also was taking issue with the syntax for some named arguments. I may take a stab at it at some point if you don't beat me to it.

**Reply by priyavadan** [reply]
yup , it is exactly that. Ruby 3 moved to be more python like with its args and kwargs

I will look at it this weekend if I can.

**Comment by priyavadan** [comment]
for lint, I believe the newer Rubocop versions had namespace changes , so the rules yml file will need to be updated (which will be painful but not too complex to do)

I restricted to rubocop v0.30 and linter works (but with a slighter older syntax as seen from the screenshot)

![image](https://user-images.githubusercontent.com/3895377/164546606-b91634d8-801d-4dd8-931d-9728146d3ac5.png)

**Reply by pderouen** [reply]
Feel free to push changes to that branch

**Reply by priyavadan** [reply]
just pushed , lint fixes and the updates from above.

**Comment by pderouen** [comment]
I tried adding ruby 3 to the regression tests. The tests fail because gem install origen is installing an ancient (0.30 something) version of origen that is locked to an incompatible version of bundler. So, origen -v is failing. I wasn't able to get gem install origen to install a newer version, not sure why. It's weird. The same happened on my ruby 3 machine.

What is your origen version outside of a workspace with your Ruby 3 install?
